com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -186,6 +202,37 @@ void OnEnable()
             LazyInitialize();
         }
 
+
+        public void OnBeforeSerialize()
+        {
+            if (maxStep != agentParameters.maxStep && !hasUpgradedFromAgentParameters)
```

Comment:
Not a problem for most users, but I think this will overwrite the Agent.maxStep for assets where we've already set it.

Maybe either
1) add `&& maxStep != 0` to the condition, or
2) `maxStep = Mathf.max(maxStep, agentParameters.maxStep);` instead - they should never both be non-zero, so that seems like the simplest approach.

Reply:
makes sense

Reply:
`maxStep == 0`